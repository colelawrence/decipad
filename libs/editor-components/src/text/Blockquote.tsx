import { PlateComponent } from '@decipad/editor-types';
import { assertElementType } from '@decipad/editor-utils';
import { Blockquote as UIBlockquote } from '@decipad/ui';
import { DraggableBlock } from '../block-management';

export const Blockquote: PlateComponent = ({
  attributes,
  children,
  element,
}) => {
  assertElementType(element, 'blockquote');

  return (
    <DraggableBlock blockKind="blockquote" element={element} {...attributes}>
      <UIBlockquote>{children}</UIBlockquote>
    </DraggableBlock>
  );
};
