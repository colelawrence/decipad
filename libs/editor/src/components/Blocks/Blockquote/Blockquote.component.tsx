import styled from '@emotion/styled';
import { SlatePluginComponent } from '@udecode/slate-plugins';
import React from 'react';

const Element = styled('div')({
  backgroundColor: '#BEE3F8',
  color: '#2B6CB0',
  lineHeight: '1.75',
  border: '1px solid #90CDF4',
  padding: '12px 24px',
  borderRadius: '8px',
  fontStyle: 'italic',
  fontSize: '16px',
  boxShadow: '0px 2px 24px -4px rgba(36, 36, 41, 0.06)',
});

export const Blockquote: SlatePluginComponent = ({ attributes, children }) => {
  return <Element {...attributes}>{children}</Element>;
};
