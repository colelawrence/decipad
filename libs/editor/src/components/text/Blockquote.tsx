import { atoms } from '@decipad/ui';
import { PlateComponent } from '../../types';

export const Blockquote: PlateComponent = ({ attributes, children }) => {
  return (
    <div {...attributes}>
      <atoms.Blockquote>{children}</atoms.Blockquote>
    </div>
  );
};
