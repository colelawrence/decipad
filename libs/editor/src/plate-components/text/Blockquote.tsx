import { types } from '@decipad/editor-config';
import { atoms } from '@decipad/ui';
import { DraggableBlock } from '../../components';

export const Blockquote: types.PlateComponent = ({
  attributes,
  children,
  element,
}) => {
  if (!element) {
    throw new Error('Blockquote is not a leaf');
  }

  return (
    <div {...attributes}>
      <DraggableBlock blockKind="blockquote" element={element}>
        <atoms.Blockquote>{children}</atoms.Blockquote>
      </DraggableBlock>
    </div>
  );
};
