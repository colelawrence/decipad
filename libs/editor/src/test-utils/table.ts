import {
  ELEMENT_TABLE,
  ELEMENT_TH,
  ELEMENT_TR,
  ELEMENT_TD,
} from '@udecode/plate';
import {
  InteractiveTable,
  TableCellType,
  Td,
  Th,
} from '../plugins/InteractiveTable/table';
import {
  ELEMENT_TABLE_CAPTION,
  ELEMENT_THEAD,
  ELEMENT_HEAD_TR,
  ELEMENT_TBODY,
} from '../utils/elementTypes';

/**
 * Creates a row with 1 named column and {@param numberOfRows} rows of non-empty cells.
 */
export const createTable = (numberOfRows: number): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType: 'string' },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: Array.from({ length: numberOfRows }, (_, i) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType: 'string' },
            children: [{ text: `Col 1 Row ${i + 1}` }],
          },
        ],
      })),
    },
  ],
});

export const createTableWithTypeAndData = (
  cellType: TableCellType,
  data: string[]
): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: data.map((text) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType },
            children: [{ text }],
          },
        ],
      })),
    },
  ],
});

export const printTable = ({
  children: [_, thead, tbody],
}: InteractiveTable): string[] => {
  const tdsToStrings = (tds: (Th | Td)[]) =>
    tds.map((td) => td.children[0].text).join(' | ');

  return [
    tdsToStrings(thead.children[0].children),
    ...tbody.children.map((tr) => tdsToStrings(tr.children)),
  ];
};
