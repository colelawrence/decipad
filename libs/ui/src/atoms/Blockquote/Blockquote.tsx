/* eslint decipad/css-prop-named-variable: 0 */
import { useIsBlockActive } from '@decipad/react-contexts';
import { css } from '@emotion/react';
import { ReactNode } from 'react';
import { cssVar, setCssVar } from '../../primitives';
import { blockAlignment } from '../../styles';

const verticalPadding = '6px';

const styles = css(
  blockAlignment.blockquote.typography,
  setCssVar('currentTextColor', cssVar('weakTextColor')),
  {
    padding: `${verticalPadding} 30px ${verticalPadding} 12px`,
    borderLeft: `4px solid ${cssVar('highlightColor')}`,
    wordBreak: 'break-word',
  }
);

const activeStyles = css({
  backgroundColor: cssVar('highlightColor'),
  boxShadow: `0px 0px 0px 100vmin ${cssVar('highlightColor')}`,
  clipPath: `inset(0 -8px 0 -8px round 8px)`,
});

interface BlockquoteProps {
  readonly children: ReactNode;
}

export const Blockquote = ({
  children,
}: BlockquoteProps): ReturnType<React.FC> => {
  const isBlockActive = useIsBlockActive();
  return (
    <blockquote
      aria-label="column-content"
      css={[styles, isBlockActive && activeStyles]}
    >
      {children}
    </blockquote>
  );
};
