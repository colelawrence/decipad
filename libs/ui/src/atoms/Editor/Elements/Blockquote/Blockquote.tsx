import { css } from '@emotion/react';
import { PlatePluginComponent } from '@udecode/plate';
import {
  blockquote,
  cssVar,
  p16Regular,
  setCssVar,
} from '../../../../primitives';

const styles = css(p16Regular, {
  ...setCssVar('currentTextColor', cssVar('weakTextColor')),
  borderLeft: `5px solid ${blockquote.rgb}`,
  padding: '8px 24px',
  margin: '16px 0',
});

export const BlockquoteElement: PlatePluginComponent = ({
  attributes,
  children,
}) => {
  return (
    <blockquote css={styles} {...attributes}>
      {children}
    </blockquote>
  );
};
