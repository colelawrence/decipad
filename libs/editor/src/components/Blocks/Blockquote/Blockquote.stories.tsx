import { LoremIpsum } from 'lorem-ipsum';
import React from 'react';
import { Blockquote } from './Blockquote.component';

export default {
  title: 'Editor/Blocks/Blockquote',
  component: Blockquote,
};

const lorem = new LoremIpsum({
  sentencesPerParagraph: {
    max: 8,
    min: 4,
  },
  wordsPerSentence: {
    max: 16,
    min: 4,
  },
});

export const Default = (args: { text: string }) => (
  <Blockquote
    leaf={{ attributes: {}, text: '' }}
    element={{ children: [{ text: '' }] }}
    text={{ text: '' }}
    attributes={{
      'data-slate-node': 'element',
      ref: null,
      'data-slate-leaf': true,
    }}
  >
    {args.text}
  </Blockquote>
);

Default.args = {
  text: lorem.generateParagraphs(5),
};

Default.storyName = 'Blockquote';
