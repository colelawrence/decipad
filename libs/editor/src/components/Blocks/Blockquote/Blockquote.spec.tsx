import { render } from 'test-utils';
import { Default } from './Blockquote.stories';

describe('Blockquote Block', () => {
  it('matches snapshot', () => {
    const { container } = render(<Default text="Hello World" />);
    expect(container).toMatchSnapshot();
  });
});
