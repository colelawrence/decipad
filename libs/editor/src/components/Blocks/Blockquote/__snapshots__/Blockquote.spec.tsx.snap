// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote Block matches snapshot 1`] = `
.emotion-0 {
  background-color: #BEE3F8;
  color: #2B6CB0;
  line-height: 1.75;
  border: 1px solid #90CDF4;
  padding: 12px 24px;
  border-radius: 8px;
  font-style: italic;
  font-size: 16px;
  box-shadow: 0px 2px 24px -4px rgba(36, 36, 41, 0.06);
}

<div>
  <div
    class="emotion-0"
    data-slate-leaf="true"
    data-slate-node="element"
  >
    Hello World
  </div>
</div>
`;
